Update images paths from ghcr.io/kubeflow/kubeflow/* to ghcr.io/kubef…

yehudit1987 opened the pull request:
This PR is a follow up PR for: 
#112 
#110 
#111 
In the previous PR's we publish the images to `ghcr.io/kubeflow/kubeflow/*` path, where this PR fix that to - `ghcr.io/kubeflow/dashboard/*` in publish as well as other use cases.

kimwnasptd commented:
ΩI think the problem here is that the current tests is that the KFAM image is never built for the `integration-tests` tag, as it was in the past.

Specifically, IIUC, what is going on is:
1. The [profiles-kfam-integration](https://github.com/kubeflow/dashboard/blob/8ce75658d1563de2a29b65e77afcf36050cd925c/.github/workflows/profiles_kfam_integration_test.yaml#L52-L60) test gets triggered
2. The test installs Profiles Manifests by using the [deploy_component.sh](https://github.com/kubeflow/dashboard/blob/8ce75658d1563de2a29b65e77afcf36050cd925c/testing/gh-actions/deploy_component.sh) script
3. That script builds the image only for the Profiles Controller
4. KFAM never gets built, and the tests are trying to pull an image that doesn't exist

This wasn't the case [in the past](https://github.com/kubeflow/kubeflow/blob/643f812f3fd0d29ef5b55f8d149ab40932cf8522/.github/workflows/profiles_kfam_integration_test.yaml#L31-L39), as tests were building both KFAM and Profiles Controller.

IIUC, right now even if we manually publish a `ghcr.io/kubeflow/dashboard/kfam:integration-tests` image, then the tests would always be using this image and will be never building and using a KFAM image as they used to.

andyatmiami commented:
Yes, I plan on reviewing this today/tomorrow.

andyatmiami reviewed (changes requested):
@yehudit1987 

First off - great work getting all these changes incoporated/aligned...

I am/will be carefully reviewing all the changes over the next day - but one thing that initially jumped out I think we'll want to address.

The `components/admission-webhook` **FOLDER** itself I think we want to rename to `poddefaults-webhooks`

I am based this off the discussion included in #118 :

<img width="976" height="637" alt="image" src="https://github.com/user-attachments/assets/99d0bb73-7418-4e31-a493-d01b37f6cdeb" />

You have already renamed the deployment - so we just wanna rename the folder to have _alignment throughout_ .

More reviewing on my end to come - but wanted to raise this first!

andyatmiami commented on .github/workflows/central_dashboard_angular_integration_test.yaml:
```diff
@@ -51,8 +51,8 @@ jobs:
     - name: Build KFAM Image
```
```suggestion
    - name: Build Access Management Image
```

andyatmiami commented on .github/workflows/centraldb_angular_docker_publish.yaml:
```diff
@@ -0,0 +1,57 @@
+      - main
+      - v*-branch
+    paths:
+      - components/dashboard-angular/**
```
```suggestion
      - components/centraldashboard-angular/**
```

andyatmiami commented on .github/workflows/centraldb_angular_docker_publish.yaml:
```diff
@@ -0,0 +1,57 @@
+      - releasing/version/VERSION
+
+env:
+  IMG: ghcr.io/kubeflow/dashboard/dashboard-angular
```
I think we should define a `COMPONENT_PATH: components/centraldashboard-angular` to centrally define the directory we want to `cd` into in subsequent steps.

ℹ️ Also please note that this file currently **incorrectly** defines this directory as `components/dashboard-angular` in subsequent steps.

andyatmiami commented on .github/workflows/centraldb_angular_docker_publish.yaml:
```diff
@@ -0,0 +1,57 @@
+
+    - name: Run dashboard-Angular build/push
+      run: |
+        cd components/dashboard-angular
```
```suggestion
        cd ${{ env.COMPONENT_PATH }}
```

see: https://github.com/kubeflow/dashboard/pull/115/files#r2288233149

andyatmiami commented on .github/workflows/centraldb_angular_docker_publish.yaml:
```diff
@@ -0,0 +1,57 @@
+      if: github.ref == 'refs/heads/main'
+      run: |
+        export TAG=latest
+        cd components/dashboard-angular
```
```suggestion
        cd ${{ env.COMPONENT_PATH }}
```

see: https://github.com/kubeflow/dashboard/pull/115/files#r2288233149

andyatmiami commented on .github/workflows/centraldb_angular_docker_publish.yaml:
```diff
@@ -0,0 +1,57 @@
+        cd components/dashboard-angular
+        make docker-build docker-push
+
+    - name: Run dashboard-Angular build/push on Version change
```
```suggestion
    - name: Run CentralDashboard-Angular build/push on Version change
```

andyatmiami commented on components/centraldashboard-angular/manifests/base/configmap.yaml:
```diff
@@ -1,7 +1,7 @@
 kind: ConfigMap
 metadata:
-  name: centraldashboard-angular-config
+  name: dashboard-angular-config
```
FYI: @juliusvonkohout @kimwnasptd @kimwnasptd 

Its not clear to me how we want to handle this configmap name.  

One on hand - the name change implemented here is consistent with the manifest renames we are undertaking.

However, i'm not quite clear on the level of "support" for `centraldashboard-angular`..
- [public documentation](https://www.kubeflow.org/docs/components/central-dash/customize/) only mentions `centraldashboard` component
- [`centraldashboard-angular` UI code ](https://github.com/kubeflow/dashboard/blob/da7645290d374590f362caa6e3fa0d9fade7dfcc/components/centraldashboard-angular/backend/app/k8s_service.ts#L5) seems to hardcode the configmap name to the `centraldashboard` component's configmap as the default/fallback(?)
- [`centraldashboard-angular` UI code](https://github.com/kubeflow/dashboard/blob/da7645290d374590f362caa6e3fa0d9fade7dfcc/components/centraldashboard-angular/backend/config/centraldashboard-config.yaml) seems to itself have a hardcoded manifest in its source.. that was already not aligned with the original name present in this file in `manifests/` directory

andyatmiami reviewed (changes requested):
https://github.com/kubeflow/dashboard/blob/da7645290d374590f362caa6e3fa0d9fade7dfcc/testing/gh-actions/test_service.sh can have a few of its `$OPERATION` `case` blocks simplified as the "default case" ( `*)` ) properly handles all the permutations itself without any other case blocks needed to be explicitly provided:
- `check_logs`
    - https://github.com/kubeflow/dashboard/blob/da7645290d374590f362caa6e3fa0d9fade7dfcc/testing/gh-actions/test_service.sh#L73
- `check_errors`
    - https://github.com/kubeflow/dashboard/blob/da7645290d374590f362caa6e3fa0d9fade7dfcc/testing/gh-actions/test_service.sh#L84
    
In both these cases - we can do away with the `case` block entirely and replace it simply with the "default case" as the `SERVICE_NAME` is consistently aligned with the `app` label selector

andyatmiami reviewed (changes requested):
I think we should remove the `cloudbuild.yaml` files present in the following locations:
- components/centraldashboard/cloudbuild.yaml
- components/access-management/cloudbuild.yaml
- components/profile-controller/cloudbuild.yaml

FYI: @kimwnasptd @juliusvonkohout @thesuperzapper 

This seems to be a (very) outdated CI/CD artifact related to GCB that is no longer required/leveraged.  As our CI/CD now leverages `ghcr` - would be good to remove these files files entirely to avoid any future confusion.  As the files also construct image name references - its particularly relevant to this PR.

yehudit1987 commented:
> ### legacy/invalid(?) files
> #### cloudbuild.yaml
> I think we should remove the `cloudbuild.yaml` files present in the following locations:
> 
> * components/centraldashboard/cloudbuild.yaml
> * components/access-management/cloudbuild.yaml
> * components/profile-controller/cloudbuild.yaml
> 
> FYI: @kimwnasptd @juliusvonkohout @thesuperzapper
> 
> This seems to be a (very) outdated CI/CD artifact related to GCB that is no longer required/leveraged. As our CI/CD now leverages `ghcr` - would be good to remove these files files entirely to avoid any future confusion. As the files also construct image name references - its particularly relevant to this PR.
> 
> #### make-win.ps1
> In a similar vein - I'm not sure what to do about this `make-win.ps1` file that contains an $IMG reference using the `gcr.io` repo as well as the now outdated image name:
> 
> * https://github.com/kubeflow/dashboard/blob/da7645290d374590f362caa6e3fa0d9fade7dfcc/components/centraldashboard/make-win.ps1#L15
> 
> This `gcr.io` `kubeflow-images-public` registry contains a lot of images that were created 5+ years ago - but interestingly seemed to be updated near on the onset of 2025 🤔
> 
> #### README.md
> And finally, the `README.md` is similarly incorrect/outdated for `centraldashboard` with references to `gcr.io` as well as label selectors that are or will be invalid upon merge of this PR
> 
> * https://github.com/kubeflow/dashboard/blob/da7645290d374590f362caa6e3fa0d9fade7dfcc/components/centraldashboard/README.md

Regarding that - 
1. I deleted the cloudbuild files since they do seems to be redundant and not relevant. 
2. As for the make-win.ps1 I updated the refences in it to the correct repo (for cases of windows users). If not needed at all it can be deleted but it's not my decision to make. 
3. I updated the README.md file.

andyatmiami commented on .github/workflows/central_dashboard_integration_test.yaml:
```diff
@@ -16,8 +16,8 @@ concurrency:
 env:
-  DASHBOARD_IMG: ghcr.io/kubeflow/dashboard/central-dashboard
-  KFAM_IMG: ghcr.io/kubeflow/dashboard/kfam
+  DASHBOARD_IMG: ghcr.io/kubeflow/dashboard/dashboard
```
There are numerous references in this file that still pass a label selector (or an argument to a script that is consumed as a label selector) that is using the `centraldashboard` name..

we should update all such occurrences to `dashboard`

juliusvonkohout commented:
@yehudit1987 @andyatmiami can we merge this soon, even if it i s not perfect ?

yehudit1987 commented:
> @yehudit1987 @andyatmiami can we merge this soon, even if it i s not perfect ? Because we are blocking many other PRs as for example [#135 (comment)](https://github.com/kubeflow/dashboard/pull/135#issuecomment-3219002876) and #133
> 
> I am willing to approve, because we need to move forward and get a release and i do not see activity from other maintainers.

Hi @juliusvonkohout and @andyatmiami as for today status is - I finished checking testing logs and fixing issues I saw. Hopefully I didn't miss anything. I believe we can merge it as is (if review is passing of course). If any other issues are remain - such as logs I miss or changing imports at go files etc. I believe we can open separate tasks for that.
